xmodule/video_block/video_block.py

```diff
@@ -472,26 +478,69 @@ def get_html(self, view=STUDENT_VIEW, context=None):  # lint-amnesty, pylint: di
             'track': track_url,
             'transcript_download_format': transcript_download_format,
             'transcript_download_formats_list': self.fields['transcript_download_format'].values,  # lint-amnesty, pylint: disable=unsubscriptable-object
-            # provide the video url iif the video is public and we are in LMS.
-            # Reverse render_public_video_xblock is not available in studio.
-            'public_video_url': self._get_public_video_url(),
         }
 
+        # Public video previewing / social media sharing
+        if self.is_public_sharing_enabled():
+            template_context['public_sharing_enabled'] = True
+            template_context['public_video_url'] = self.get_public_video_url()
+
         return self.runtime.service(self, 'mako').render_template('video.html', template_context)
 
-    def _get_public_video_url(self):
+    def get_course_video_sharing_override(self):
         """
-        Returns the public video url
+        Return course video sharing options override or None
+        """
+        course = get_course_by_id(self.course_id)
```

Comment:
Should you worry about the possible null `course` variable? Maybe we should handle that error and log error, but return false?

Reply:
Ugh, I officially hate this function. Guess what it returns on failure? An `Http404` 😒 

Which warrants a broad exception, I think. I'm not convinced we'll ever get this as an error but if it makes you happy...